spyder/plugins/editor.py

```diff
@@ -2643,3 +2641,23 @@ def set_create_new_file_if_empty(self, value):
         """Change the value of create_new_file_if_empty"""
         for editorstack in self.editorstacks:
             editorstack.create_new_file_if_empty = value
+
+    def move_editorstack_data(self, start, end):
+        """
+        Move editorstack.data to be synchronized when tabs are moved.
+        """
+        if start < 0 or end < 0:
+            return
+        else:
+            steps = abs(end - start)
+            sign = (end - start) // steps
+
+            for editorstack in self.editorstacks :
+                data = editorstack.data
+                editorstack.blockSignals(True)
+
+                for i in range(start, end, sign*1):
+                    data[i], data[i+sign] = data[i+sign], data[i]

```

Comment:
It would be probably better to invert the position of the for so that we move all editors at the same time 

```python

for i in range(start, end, sign * 1):
    for editorstack in self.editorstacks :
        data = editorstack.data
        editorstack.blockSignals(True)

        data[i], data[i + sign] = data[i + sign], data[i]

        editorstack.blockSignals(False)
        editorstack.refresh()
```

It feels that in case there is an error it would leave the UI in a better state.

Thoughts @rlaverde, @ccordoba12 ?